vars/envToPragmas.groovy

```diff
@@ -0,0 +1,18 @@
+/* groovylint-disable ParameterName */
+// vars/envToPragmas.groovy
+
+/**
+ * envToPragmas.groovy
+ *
+ * envToPragmas variable
+ */
+
+Map call() {
+    Map pragmas = [:]
+    if (env.pragmas)
+        pragmas = "${env.pragmas}"[1..-2].split(', ').collectEntries { entry ->
+            String[] pair = entry.split('=', 2)
+            [(pair.first().trim()): pair.last().trim()]
+    }
+    return pragmas
+}
```

Comment:
@brianjmurrell The previous version of this was doing
```
Map pragmas = "${env.pragmas}"[1..-2].split(', ').collectEntries { entry ->
            String[] pair = entry.split('= ')
            [(pair.first()): pair.last()]
        }
```
Which implicitly strips the first space from the value. I.e. it splits on `"= "`
This new version should still work similar, except it also allows the value to not start with a space.
But for some reason I've yet to explain, I think some other version of this this function is being used when a `daos` PR runs.

For example, in my `daos` PR, in the get commit message stage:
```
env.pragmas after updating
{skip-build=true, skip-test=true, skip-tests=true, required-githooks=true, signed-off-by=Dalton Bohning <dalton.bohning@intel.com>, test-tag= DfuseMUPerms DmvrDstCreate pool pr}
```

Notice there is NOT a space after the equals sign. That's because `envToPragmas` strips it, and then when the Map is converted back to a string in `env.pragmas`, the space is gone.

The problem is when we then try to parse that `env.pragmas` a second time. I did verify with a unit test that this code can parse and update multiple times, but on my `daos` PR, the `skip-build:true` pragma is being ignored.

I suspect we're actually calling some other version of this function. Maybe the one in `system-pipeline-lib`?

Reply:
I've tested this code locally and in the new unit test and both behave as expected.

Reply:
@brianjmurrell, @phender and I added more debugging and confirmed that a `daos` PR is in fact using `cachedCommitPragma` from `system-pipeline-lib`, NOT `pipeline-lib`